tests/test_apis.py

```
@@ -74,16 +76,49 @@ def test_api(group_name, test_name, case, http_session):
+                    try:
+                        assert (
+                            key in response_dict
+                            and response_dict[key] == value
+                        )
+                    except AssertionError:
+                        error_detail = find_first_error(
+                            value, response_dict.get(key)
+                        )
+                        raise AssertionError(
+                            f"Expected {key} to be {value}, but got {response_dict.get(key)}. \n\nDetail: {error_detail}"
+                        )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)